avm/ptn/lz/sub-vending/main.bicep

```diff
@@ -0,0 +1,750 @@
+
+- Type: `[]` Array
+- Default value: `[]` *(empty array)*.
+''')
+param virtualNetworkVwanPropagatedLabels array = []
+
+@metadata({
+  example: false
+})
+@description('''Optional. Indicates whether routing intent is enabled on the Virtual Hub within the Virtual WAN.
+
+- Type: Boolean.
+''')
+param vHubRoutingIntentEnabled bool = false
+
+@metadata({
+  example: true
+})
+@description('''Optional. Whether to create role assignments or not. If true, supply the array of role assignment objects in the parameter called `roleAssignments`.
+
+- Type: Boolean.
+''')
+param roleAssignmentEnabled bool = false
+
+@metadata({
+  example: [
+    {
+      principalId: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
+      definition: 'Contributor'
+      relativeScope: ''
+    }
+    {
+      principalId: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
+      definition: '/providers/Microsoft.Authorization/roleDefinitions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
+      relativeScope: ''
+    }
+    {
+      principalId: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
+      definition: 'Reader'
+      relativeScope: '/resourceGroups/rsg-networking-001'
+    }
+    {
+      principalId: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
+      definition: '/providers/Microsoft.Authorization/roleDefinitions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
+      relativeScope: '/resourceGroups/rsg-networking-001'
+    }
+  ]
+})
+@description('''Optional. Supply an array of objects containing the details of the role assignments to create.
+
+Each object must contain the following `keys`:
+- `principalId` = The Object ID of the User, Group, SPN, Managed Identity to assign the RBAC role too.
+- `definition` = The Name of one of the pre-defined built-In RBAC Roles or a Resource ID of a Built-in or custom RBAC Role Definition as follows:
+  - You can only provide the RBAC role name of the pre-defined roles (Contributor, Owner, Reader, Network Contributor, Role Based Access Control Administrator (Preview), User Access Administrator, Security Admin). We only provide those roles as they are the most common ones to assign to a new subscription, also to reduce the template size and complexity in case we define each and every Built-in RBAC role.
+  - You can provide the Resource ID of a Built-in or custom RBAC Role Definition
+    - e.g. `/providers/Microsoft.Authorization/roleDefinitions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`
+  - You can provide the RBAC role Id of a Built-in RBAC Role Definition
+    - e.g. `xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`
+- `relativeScope` = 2 options can be provided for input value:
+    1. `''` *(empty string)* = Make RBAC Role Assignment to Subscription scope
+    2. `'/resourceGroups/<RESOURCE GROUP NAME>'` = Make RBAC Role Assignment to specified Resource Group
+
+> See below [example in parameter file](#parameter-file) of various combinations
+
+- Type: `[]` Array
+- Default value: `[]` *(empty array)*.
+''')
```

Comment:
The type & default value should not be in the description as it's added automatically by the tooling in the readme.

Also, you did a grat job describing the parameter - so much so that I can't help but wonder if it would not be easiert to have User Defined Type instead.

Reply:
Mind you, the feedback regarding the type & default value applies to all parameters

Reply:
Example (how it is now) : 
![image](https://github.com/Azure/bicep-registry-modules/assets/5365358/fef1de48-88f7-476b-99af-10a3220d8818)